From 54ab0b36c9f206dde1d8c33b156673e1424a3fc0 Mon Sep 17 00:00:00 2001
From: Hanna Hawa <hannah@marvell.com>
Date: Thu, 3 Nov 2016 18:06:29 +0200
Subject: [PATCH 0558/1345] mmc: a80x0: a70x0: enable MMC/SDIO interface in
 CP110

commit  84f461b92c129b2481720814ce262986c8a9afae from
https://github.com/MarvellEmbeddedProcessors/linux-marvell.git

enable eMMC/SDIO interface in CP110 for Armada-70x0/Armada-80x0

Change-Id: I6d6e1c1801ae31c838d407da0255e58f8523c7f5
Signed-off-by: Hanna Hawa <hannah@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/33559
Reviewed-by: Nadav Haklai <nadavh@marvell.com>
Tested-by: Star_Automation <star@marvell.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 .../boot/dts/marvell/armada-7040-db-all-pcie.dts   |    3 +++
 .../arm64/boot/dts/marvell/armada-7040-db-nand.dts |    3 +++
 .../boot/dts/marvell/armada-7040-db-router.dts     |    3 +++
 arch/arm64/boot/dts/marvell/armada-7040-db.dtsi    |    3 +++
 .../boot/dts/marvell/armada-8040-db-router.dts     |    3 +++
 arch/arm64/boot/dts/marvell/armada-8040-db.dtsi    |    3 +++
 6 files changed, 18 insertions(+)

diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-all-pcie.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-all-pcie.dts
index 529a5c8..3691bf0 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-all-pcie.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-all-pcie.dts
@@ -100,6 +100,9 @@
 			udc@524100 {
 				status = "disabled";
 			};
+			sdhci@780000 {
+				status = "okay";
+			};
 			spi@700680 {
 				status = "okay";
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts
index 59af84d..8942690 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-nand.dts
@@ -100,6 +100,9 @@
 			udc@524100 {
 				status = "disabled";
 			};
+			sdhci@780000 {
+				status = "okay";
+			};
 			nand@720000 {
 				status = "okay";
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts b/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts
index 10f8a3b..f0172cd 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db-router.dts
@@ -100,6 +100,9 @@
 			udc@524100 {
 				status = "disabled";
 			};
+			sdhci@780000 {
+				status = "okay";
+			};
 			spi@700680 {
 				status = "okay";
 			};
diff --git a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
index 0ab7c5b..53fe9d6 100644
--- a/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-7040-db.dtsi
@@ -103,6 +103,9 @@
 			usb3h1: usb3@510000 {
 				usb-phy = <&usb3h1_phy>;
 			};
+			sdhci@780000 {
+				non-removable;
+			};
 			spi@700680 {
 				spi-flash@0 {
 					#address-cells = <0x1>;
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts b/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
index bbcf7f4..e763e2a 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db-router.dts
@@ -67,6 +67,9 @@
 			usb3h1: usb3@510000 {
 				status = "okay";
 			};
+			sdhci@780000 {
+				status = "okay";
+			};
 			mdio@12a200 {
 				#address-cells = <1>;
 				#size-cells = <0>;
diff --git a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
index d5052ec..7d67961 100644
--- a/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
+++ b/arch/arm64/boot/dts/marvell/armada-8040-db.dtsi
@@ -152,6 +152,9 @@
 			usb3@510000 {
 				usb-phy = <&cpm_usb3h1_phy>;
 			};
+			sdhci@780000 {
+				non-removable;
+			};
 		};
 
 		cpm_usb3h0_phy: cpm_usb3_phy0 {
-- 
1.7.9.5

